mslib/_tests/utils.py

```diff
@@ -153,6 +158,39 @@ def mscolab_get_project_id(app, msc_url, email, password, username, path):
             return p['p_id']
 
 
+def mscolab_check_free_port(all_ports, port):
+    _s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        _s.bind(("127.0.0.1", port))
+    except (socket.error, IOError):
+        port = all_ports.pop()
+        mscolab_check_free_port(all_ports, port)
+    _s.close()
+    return port
+
+
+def mscolab_start_server(all_ports, mscolab_settings=mscolab_settings):
+    handle_db_seed()
+    port = mscolab_check_free_port(all_ports, all_ports.pop())
```

Comment:
yes, this is a fast solution to get a port for checking and kick it from the list.

the parallel tests can be run by 

--dist loadscope: Tests are grouped by module for test functions and by class for test methods. Groups are distributed to available workers as whole units. This guarantees that all tests in a group run in the same process. This can be useful if you have expensive module-level or class-level fixtures. Grouping by class takes priority over grouping by module.
--dist loadfile: Tests are grouped by their containing file. Groups are distributed to available workers as whole units. This guarantees that all tests in a file run in the same worker.